File: 은지/LV2/다음 큰 숫자.py
def two(x):
    if x < 2:
        return x
    else:
        return x % 2 + two(x//2) * 10

def solution(n):
    answer = 0
    two_n = str(two(n))
    m = two_n.count("1")
    for x in range(n+1, 1000001):
        if str(two(x)).count("1") == m:
            answer = x
            break
    return answer

n = 78
print(solution(n))

File: 은지/LV2/다음 큰 숫자.js
function count(s) {
  var cnt = 0
  for (let v of s) {
    if (v === "1") {
      cnt += 1
    }
  }
  return cnt
}

function solution(n) {
  var answer = 0;
  var m = count(n.toString(2))  
  for (let x=n+1; x<1000001; x++) {
    if (count(x.toString(2)) === m) {
      answer = x
      break
    }
  }
  return answer;
}